re/service.go

```diff
@@ -112,23 +113,27 @@ type Service interface {
 }
 
 type re struct {
-	idp    supermq.IDProvider
-	repo   Repository
-	pubSub messaging.PubSub
-	errors chan error
-	ticker Ticker
-	email  Emailer
-	ts     []transformers.Transformer
+	writersPubSub messaging.PubSub
+	alarmsPubSub  messaging.PubSub
+	rePubSub      messaging.PubSub
```

Comment:
Let's use appropriate structs: we need only publishers for alarms and writers for the time being (subs hold heavy maps that we'd unnecessarily drag around with pubSub) - `writersPub` and `alramsPub`. For RE, keep pubsub, but rename it to `rulesPubSub`.